### PR TITLE
scripts: build.sh for proper `zig build` on macOS Ventura

### PR DESCRIPTION
**Pull request by kprotty**
On macOS Ventura (13), building an exe when the OS portion of the target is "native" (i.e. `aarch64-native`, NOT `native-macos`) hits an abort in zig's compiler on 0.9. This abort was fixed in 0.10, but we are waiting until 0.11 to update for now. In the meantime, a workaround is required.

`zig build` first generates an exe from `build_runner.zig` which imports our `build.zig`. Then, it runs this exe with our build flags. Said exe is built with a "native" OS portion of the target and hits the abort on 0.9 well before touching our build.zig.

Instead, we have our own copy of `build_runner.zig` and run it with `-target native-macos` on Ventura which doesn't hit the abort. We also makes sure to pass the correct `-Dtarget` in `benchmark.sh`.

## Pre-merge checklist

Performance:

* [x] I am very sure this PR could not affect performance.

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -26,13 +26,17 @@ if [ ! -d "zig" ]; then
...
 if [ "$debug" = "true" ]; then
     echo "Building Tigerbeetle debug..."
-    # shellcheck disable=SC2086
-    zig/zig build install -Dcpu=baseline ${target}
+    ./scripts/build.sh install -Dcpu=baseline "${target}"
 else
     echo "Building TigerBeetle..."
-    # shellcheck disable=SC2086
-    zig/zig build install -Dcpu=baseline -Drelease-safe ${target}
+    ./scripts/build.sh install -Dcpu=baseline -Drelease-safe "${target}"
```
Comment (eatonphil) [inline scripts/install.sh]: @cb22 do we need to not quote $target here?

File: scripts/build.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (eatonphil) [inline scripts/build.sh]: Ah, this can't be bash :) It needs to be sh

File: scripts/install.sh (hunk 1 above)
Comment (kprotty) [reply scripts/install.sh]: It was recommended to me from shellcheck

**Review by eatonphil — APPROVED** [review]
Amazing, thanks King! I've tested this on my Ventura VM and my non-Ventura non-VM.

**Comment by BenSchZA** [comment]
Can confirm building from source and demo work on my Ventura M1 laptop 🚀 

image

**Comment by eatonphil** [comment]
@kprotty can we merge this?

**Comment by kprotty** [comment]
yup, good to go @eatonphil

**Pull request merged**